install.sh

```
@@ -1238,92 +1281,310 @@ update_zshrc() {
+# Load all available presets into arrays
+load_available_presets() {
+    AVAILABLE_PRESETS=()
+    AVAILABLE_PRESET_DESCRIPTIONS=()
+    AVAILABLE_PRESET_PACKAGES_MAP=()
+
+    # Load from external files if available
+    local found_files=false
+    for preset_file in "$SCRIPT_DIR"/presets/*.sh; do
+        [ -f "$preset_file" ] || continue
+        found_files=true
+        source "$preset_file"
```

Shellcheck (0.11.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)